core/src/main/java/org/apache/spark/unsafe/map/HashMapGrowthStrategy.java

```diff
@@ -30,11 +30,15 @@
   HashMapGrowthStrategy DOUBLING = new Doubling();
 
   class Doubling implements HashMapGrowthStrategy {
+
+    private static final int ARRAY_MAX = Integer.MAX_VALUE - 8;
```

Comment:
Maybe worth adding a comment why this values is chosen as the max
Like here
http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/util/ArrayList.java#l223